ap/selection/test.py

```diff
@@ -14,17 +14,8 @@
 np = maybe_import("numpy")
 
 
-def extract(array, idx):
-    """
-    inputs jagged array and index and returns padded array from given index
-    """
-    array = ak.pad_none(array, idx + 1)
-    array = ak.fill_none(array[:, idx], -999)
-    return array
-
-
 # object definitions
-# TODO: Producer instead of Selector
+# TODO: return masks instead of indices and build indices as late as possible
 @selector(uses={"Jet.pt", "Jet.eta"})
 def req_jet(events):
     mask = (events.Jet.pt > 30) & (abs(events.Jet.eta) < 2.4)
```

Comment:
Just as a side note: this is also the point where the bug in the selection (e.g. selecting jets with pt<30GeV) was from. The argsort also needs to be applied on the mask, e.g. like that:

https://github.com/mafrahm/analysis_playground/blob/fix_object_req_argsort/ap/selection/test.py#L28-L29

Reply:
Yep, good point!

@pkausw Was planning to change the selectors that currently return indices to return masks again, and then apply them in the correct way you suggest in a consistent manner :)